pkg/transformers/namespace.go

```diff
@@ -44,6 +44,9 @@ var skipNamespacePathConfigs = []PathConfig{
 		GroupVersionKind: &schema.GroupVersionKind{
 			Kind: "ClusterRoleBinding",
 		},
+		// The namespace pointed by this Path will be updated
+		Path:               []string{"subjects", "namespace"},
+		CreateIfNotPresent: false,
 	},
```

Comment:
discussed offline, we will mutate (update the namespace field in the resource payload) as a separate step from `skipping`.

Reply:
I added this as a final step in the transformation. I also addressed some other issues in the change.

Since the `subjects` are actually referencing to other objects, we need to update the name and namespace when those objects are updated. So here, I added a checking to see if kustomize can find a reference object. If it can, update the namespace. Otherwise, the referenced objects are external objects to current kustomization. In this case, kustomize doesn't update the namespace.